plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -270,22 +266,15 @@ public static ReaderPageSource createPageSource(
             checkArgument(column.getColumnType() == REGULAR, "column type must be REGULAR: %s", column);
         }
 
-        List<Optional<org.apache.parquet.schema.Type>> parquetFields = baseColumns.stream()
-                .map(column -> getParquetType(column, fileSchema, useColumnNames))
-                .map(Optional::ofNullable)
-                .collect(toImmutableList());
         ImmutableList.Builder<Type> trinoTypes = ImmutableList.builder();
         ImmutableList.Builder<Optional<Field>> internalFields = ImmutableList.builder();
-        for (int columnIndex = 0; columnIndex < baseColumns.size(); columnIndex++) {
-            HiveColumnHandle column = baseColumns.get(columnIndex);
-            Optional<org.apache.parquet.schema.Type> parquetField = parquetFields.get(columnIndex);
-
+        for (HiveColumnHandle column : baseColumns) {
             trinoTypes.add(column.getBaseType());
-
-            internalFields.add(parquetField.flatMap(field -> {
-                String columnName = useColumnNames ? column.getBaseColumnName() : fileSchema.getFields().get(column.getBaseHiveColumnIndex()).getName();
-                return constructField(column.getBaseType(), lookupColumnByName(messageColumn, columnName));
-            }));
+            internalFields.add(Optional.ofNullable(getParquetType(column, fileSchema, useColumnNames))
+                    .flatMap(field -> {
```

Comment:
`.flatMap` is used here "incorrectly," as a way to do `ifPresent` check and it's not real mapping.

pre-existing, so don't change (at least not in this commit)

Reply:
The only thing about that that really looks incorrect to me is that `getParquetType` doesn't already return an `Optional`.